pkg/crypto/operator/box/box.go

```diff
@@ -29,12 +29,26 @@ import (
 //   for encryption/decryption, so clients do not need to see
 //   the secrets themselves.
 type CryptoBox struct {
-	w *BaseWallet
+	w operator.KeyHolder
 }
 
-// NewCryptoBox creates a CryptoBox which provides crypto box encryption using the given wallet's keypairs
-func NewCryptoBox(w *BaseWallet) *CryptoBox {
-	return &CryptoBox{w: w}
+// New creates a CryptoBox which provides crypto box encryption using the given KeyHolder's keypairs
+func New(w wallet.Crypto) (*CryptoBox, error) {
+	b := CryptoBox{}
+	err := w.AttachCryptoOperator(&b)
```

Comment:
I'm not really sure of the purpose of this pattern.

If any package can simply request to have the operator "injected", no real implementation is being hidden from outside the framework.

Reply:
This had value when my PR limited CryptoOperators to being defined in the framework itself in `pkg/crypto` - since we need to open that up then this doesn't really have a purpose.

Reply:
As discussed, think of this functionality as the sw implementation of the crypto interface.

something like pkg/crypto/sw could make sense as a parent folder.